bitwarden_license/bit-web/src/app/secrets-manager/service-accounts/access/access.service.ts

```diff
@@ -0,0 +1,127 @@
+import { AccessTokenView } from "../models/view/access-token.view";
+
+@Injectable({
+  providedIn: "root",
+})
+export class AccessService {
+  private _accessTokenVersion = "0";
+  protected _accessToken: Subject<AccessTokenView> = new Subject();
+
+  accessToken$ = this._accessToken.asObservable();
+
+  constructor(
+    private cryptoService: CryptoService,
+    private apiService: ApiService,
+    private cryptoFunctionService: CryptoFunctionService,
+    private encryptService: EncryptService
+  ) {}
+
+  async getAccessTokens(
+    organizationId: string,
+    serviceAccountId: string
+  ): Promise<AccessTokenView[]> {
+    const r = await this.apiService.send(
+      "GET",
+      "/service-accounts/" + serviceAccountId + "/access-tokens",
+      null,
+      true,
+      true
+    );
+    const results = new ListResponse(r, AccessTokenResponse);
+
+    return await this.createAccessTokenViews(organizationId, results.data);
+  }
+
+  async createAccessToken(
+    organizationId: string,
+    serviceAccountId: string,
+    accessTokenView: AccessTokenView
+  ): Promise<string> {
+    const encryptionKey = await this.makeAccessTokenEncryptionKey();
+    const request = await this.createAccessTokenRequest(
+      organizationId,
+      encryptionKey,
+      accessTokenView
+    );
+    const r = await this.apiService.send(
+      "POST",
+      "/service-accounts/" + serviceAccountId + "/access-tokens",
+      request,
+      true,
+      true
+    );
+    const result = new AccessTokenCreationResponse(r);
+    this._accessToken.next(null);
+    // return access-token schema: [version].[service-account-id].[client-secret]:[encryption-key]
+    return `${this._accessTokenVersion}.${serviceAccountId}.${result.clientSecret}:${encryptionKey.keyB64}`;
+  }
+
+  private async createAccessTokenRequest(
+    organizationId: string,
+    encryptionKey: SymmetricCryptoKey,
+    accessTokenView: AccessTokenView
+  ): Promise<AccessTokenRequest> {
+    const organizationKey = await this.getOrganizationKey(organizationId);
+    const accessTokenRequest = new AccessTokenRequest();
+    const [name, encryptedPayload, key] = await Promise.all([
+      await this.encryptService.encrypt(accessTokenView.name, organizationKey),
+      await this.encryptService.encrypt(organizationKey.toJSON.toString(), encryptionKey),
+      await this.encryptService.encrypt(encryptionKey.keyB64, organizationKey),
+    ]);
+
+    accessTokenRequest.name = name;
+    accessTokenRequest.encryptedPayload = encryptedPayload;
+    accessTokenRequest.key = key;
+    accessTokenRequest.expireAt = accessTokenView.expireAt;
+    return accessTokenRequest;
+  }
+
+  private async makeAccessTokenEncryptionKey(numberOfBytes = 16): Promise<SymmetricCryptoKey> {
+    const keyMaterial = await this.cryptoFunctionService.randomBytes(numberOfBytes);
+    const key = await this.cryptoFunctionService.hkdf(
+      keyMaterial,
+      "bitwarden-sm-access-token",
```

Comment:
I believe this is the salt we used in the SDK.

```suggestion
      "bitwarden-accesstoken",
```

Reply:
resolved in https://github.com/bitwarden/clients/pull/4095/commits/75c11ef0efe3b4ae1c2c74ab8b99cf72ebf697fb